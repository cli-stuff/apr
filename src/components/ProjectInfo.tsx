import type { JSONSchemaForNPMPackageJsonFiles } from '@schemastore/package'
import { Box, Newline, Text } from 'ink'
import Divider from 'ink-divider'
import Link from 'ink-link'
import DetailItem from './DetailItem'

const ProjectInfo = ({ pkg }: { pkg: JSONSchemaForNPMPackageJsonFiles }) => (
	<Box
		borderStyle="round"
		minWidth={50}
		paddingLeft={1}
		paddingRight={1}
		flexDirection="column"
		alignSelf="flex-start"
	>
		<Box
			alignItems="center"
			alignSelf="center"
			flexDirection="column"
			justifyContent="center"
		>
			<Text>
				<Text bold>{pkg.name}</Text>
				<Newline />
			</Text>
			<Text>{pkg.description}</Text>
		</Box>

		<Divider />

		{pkg.version && <DetailItem label="🔖 Version">{pkg.version}</DetailItem>}
		{pkg.author?.name && (
			<DetailItem label="👤 Author">
				{pkg.author.name}
				{pkg.author.url && ` (${pkg.author.url})`}
			</DetailItem>
		)}
		{pkg.license && <DetailItem label="📄 License">{pkg.license}</DetailItem>}
		{pkg.homepage && (
			<DetailItem label="🏠 Homepage">{pkg.homepage}</DetailItem>
		)}
		{pkg.bugs && (
			// @ts-expect-error
			<DetailItem label="🐛 Bugs">{pkg.bugs?.url}</DetailItem>
		)}
	</Box>
)

export default ProjectInfo
