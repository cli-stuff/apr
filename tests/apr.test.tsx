import { expect, test } from 'bun:test'
import type { JSONSchemaForNPMPackageJsonFiles } from '@schemastore/package'
import { render } from 'ink-testing-library'
import ProjectInfo from '../src/components/ProjectInfo'

/**
 * The dummy package.json file from the playground directory.
 */ // @ts-expect-error
const pkg: JSONSchemaForNPMPackageJsonFiles = await import(
	'./playground/package.json',
	{
		assert: { type: 'json' },
	}
)

test('just works', () => {
	const { lastFrame } = render(<ProjectInfo pkg={pkg} />)
	expect(lastFrame()).toMatchSnapshot()
})
