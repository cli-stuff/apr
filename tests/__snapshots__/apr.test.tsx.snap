// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`just works 1`] = `
"╭────────────────────────────────────────────────────────────╮
│                       \x1B[1mdummy-package\x1B[22m                        │
│                                                            │
│ 🎲 A randomly generated dummy package for testing purposes │
│ \x1B[90m──────────────────────────────────────────────────────────\x1B[39m │
│ \x1B[1m\x1B[32m🔖 Version\x1B[39m\x1B[22m: 1.2.3                                          │
│ \x1B[1m\x1B[32m👤 Author\x1B[39m\x1B[22m: Test Developer (https://example.com)            │
│ \x1B[1m\x1B[32m📄 License\x1B[39m\x1B[22m: Apache-2.0                                     │
│ \x1B[1m\x1B[32m🏠 Homepage\x1B[39m\x1B[22m: https://github.com/test/dummy-package         │
│ \x1B[1m\x1B[32m🐛 Bugs\x1B[39m\x1B[22m: https://github.com/test/dummy-package/issues      │
╰────────────────────────────────────────────────────────────╯"
`;
